File: castplay-admin/src/api/media.ts
/**
 * Media API
 */
import client from './client';

export interface MediaFile {
  id: number;
  file_name: string;
  file_type: 'image' | 'video' | 'ppt';
  file_path: string;
  converted_path: string | null;
  file_size: number;
  duration: number | null;
  thumbnail_path: string | null;
  md5_hash: string | null;
  upload_time: string;
  status: 'processing' | 'ready' | 'failed';
  folder_id: number | null;
  folder_name: string | null;
}

export interface MediaListParams {
  page?: number;
  per_page?: number;
  file_type?: 'image' | 'video' | 'ppt';
  status?: 'processing' | 'ready' | 'failed';
  folder_id?: number | string | null;  // 'null' 表示根目录
}

export interface MediaListResponse {
  media: MediaFile[];
  total: number;
  page: number;
  per_page: number;
  pages: number;
}

export const mediaApi = {
  /**
   * 上传媒体文件
   */
  upload: (file: File, fileType: 'image' | 'video' | 'ppt', folderId?: number | null): Promise<{ message: string; media: MediaFile }> => {
    const formData = new FormData();
    formData.append('file', file);
    formData.append('file_type', fileType);
    if (folderId !== undefined && folderId !== null) {
      formData.append('folder_id', String(folderId));
    }

    return client.post('/media/upload', formData, {
      headers: {
        'Content-Type': 'multipart/form-data',
      },
    });
  },

  /**
   * 媒体列表
   */
  list: (params?: MediaListParams): Promise<MediaListResponse> => {
    return client.get('/media', { params });
  },

  /**
   * 媒体详情
   */
  get: (id: number): Promise<MediaFile> => {
    return client.get(`/media/${id}`);
  },

  /**
   * 删除媒体
   */
  delete: (id: number): Promise<{ message: string }> => {
    return client.delete(`/media/${id}`);
  },

  /**
   * 获取下载URL
   */
  getDownloadUrl: (id: number) => {
    return `/api/media/${id}/download`;
  },

  /**
   * 获取缩略图URL
   */
  getThumbnailUrl: (id: number) => {
    return `/api/media/${id}/thumbnail`;
  },
};

File: castplay-server/app/schemas/media.py
"""
Pydantic Schemas for Media API
"""
from datetime import datetime
from typing import Optional, List

from pydantic import BaseModel, Field


class MediaBase(BaseModel):
    """媒体基础模型"""
    filename: str = Field(..., max_length=255, description="文件名")
    file_type: str = Field(..., max_length=50, description="文件类型")


class MediaUploadResponse(BaseModel):
    """媒体上传响应"""
    message: str
    media: "MediaResponse"


class MediaResponse(BaseModel):
    """媒体响应"""
    id: int
    filename: str
    original_filename: str
    file_type: str
    file_size: int
    file_path: str
    converted_path: Optional[str] = None
    thumbnail_path: Optional[str] = None
    md5_hash: Optional[str] = None
    status: str
    folder_id: Optional[int] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class MediaListResponse(BaseModel):
    """媒体列表响应"""
    media: List[MediaResponse]
    total: int
    page: int
    per_page: int
    pages: int


class MediaUpdate(BaseModel):
    """更新媒体请求"""
    filename: Optional[str] = Field(None, max_length=255)
    folder_id: Optional[int] = None


# 解决前向引用
MediaUploadResponse.model_rebuild()
